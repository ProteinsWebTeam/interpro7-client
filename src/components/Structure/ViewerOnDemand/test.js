// @flow
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import { ViewerOnDemand } from '.';

const renderer = new ShallowRenderer();

describe('<ViewerOnDemand />', () => {
  test('should render', () => {
    renderer.render(
      <ViewerOnDemand
        id={'102m'}
        matches={[
          {
            metadata: {
              accession: 'G3DSA:1.10.490.10',
              go_terms: null,
              integrated: 'IPR012292',
              member_databases: null,
              name: 'Globins',
              source_database: 'cathgene3d',
              type: 'homologous_superfamily',
            },
            structures: [
              {
                accession: '102m',
                chain: 'A',
              },
            ],
          },
        ]}
        userActivatedVisible={false}
        changeSettingsRaw={() => {}}
      />,
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
