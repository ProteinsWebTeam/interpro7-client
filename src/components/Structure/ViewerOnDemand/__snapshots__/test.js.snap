// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ViewerOnDemand /> should render 1`] = `
<div
  className="wrapper"
>
  <button
    className="inner-wrapper"
    onClick={[Function]}
  >
    <div
      className="background"
    >
      <LazyImage
        alt="structure with accession 102m"
        src="//www.ebi.ac.uk/thornton-srv/databases/pdbsum/102m/traces.jpg"
      />
    </div>
    <div
      className="text"
    >
      <p>
        click or tap this area to display the interactive structure viewer
      </p>
      <p>
        <label
          onClick={[Function]}
        >
          Remember this next time?
           
          <input
            type="checkbox"
          />
        </label>
        <small>
          This can be reversed in the
           
          <Memo(ConnectFunction)
            to={
              Object {
                "description": Object {
                  "other": Array [
                    "settings",
                  ],
                },
              }
            }
          >
            Settings
          </Memo(ConnectFunction)>
           
          page
        </small>
      </p>
    </div>
  </button>
</div>
`;
